splunk_connect_for_snmp/inventory/tasks.py

```diff
@@ -222,3 +255,91 @@ def is_smart_profile_valid(profile_name, profile):
         logger.warning(f"Patterns for profile {profile_name} must be a list")
         return False
     return True
+
+
+def filter_condition_on_database(mongo_client, address: str, conditions: list):
+    attributes = mongo_client.attributes
+    query = create_query(conditions, address)
+    result = attributes.find(
+        query, {"address": 1, "group_key_hash": 1, "_id": 0, "indexes": 1}
+    )
+    return list(result)
+
+
+def create_profile(profile_name, varBinds, records):
```

Comment:
I am wondering whether:
def create_profile(profile_name, varBinds, records):
    varbind_copy = varBinds.copy()
    indexes_copy = [record["indexes"] for record in records]
    varbind_list = [vb + idx for idx in indexes_copy for vb in varbind_copy]
    profile = {profile_name: {"frequency": 10, "varBinds": varbind_list}}
    return profile
is a bit more optimal:
1. list comprehension instead of for loop. it is faster
2. dictionary compehension instead of empty dict and then adding values. it should be a bit faster as well
3. In the current inner loop the varbind var is copy in each iteration of this loop. I think it will be faster if we copy this outisde of the loop.

What do you think?

Reply:
I use copy of varBinds element, not the varBinds itself, so can't copy it before the loop. But, you're right I can use list comprehension because I don't even remember why the copy operator is here - probably I used it before in some other way and forgot to delete it later on. Also, you found out how I terribly hardcoded frequency 🙈 Thanks!